test/lint/lint-circular-dependencies.sh

```diff
@@ -92,7 +92,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "llmq/signing -> masternode/node -> validationinterface -> llmq/signing"
     "evo/mnhftx -> validation -> evo/mnhftx"
     "evo/deterministicmns -> validation -> evo/deterministicmns"
-    "evo/chainhelper -> masternode/payments -> validation -> evo/chainhelper"
+    "evo/chainhelper -> evo/specialtxman -> validation -> evo/chainhelper"
```

Comment:
it seems as `validation.h` is not needed in specialtxman
```diff
diff --git a/src/evo/specialtxman.cpp b/src/evo/specialtxman.cpp
index 84a4f6afef..ffb1c3d5cb 100644
--- a/src/evo/specialtxman.cpp
+++ b/src/evo/specialtxman.cpp
@@ -17,7 +17,6 @@
 #include <llmq/blockprocessor.h>
 #include <llmq/commitment.h>
 #include <primitives/block.h>
-#include <validation.h>
 
 static bool CheckSpecialTxInner(const CTransaction& tx, const CBlockIndex* pindexPrev, const CCoinsViewCache& view, const std::optional<CRangesSet>& indexes, bool check_sigs, TxValidationState& state)
 {
```

Reply:
Removing it causes the number of circular dependencies to explode

```
dash@1914d5152e74:/src/dash$ ./test/lint/lint-circular-dependencies.sh
A new circular dependency in the form of "evo/assetlocktx -> validation -> evo/specialtxman -> evo/assetlocktx" appears to have been introduced.

A new circular dependency in the form of "evo/cbtx -> validation -> evo/specialtxman -> evo/cbtx" appears to have been introduced.

A new circular dependency in the form of "evo/chainhelper -> masternode/payments -> validation -> evo/chainhelper" appears to have been introduced.

A new circular dependency in the form of "evo/creditpool -> validation -> evo/specialtxman -> evo/creditpool" appears to have been introduced.

A new circular dependency in the form of "evo/deterministicmns -> validation -> evo/specialtxman -> evo/deterministicmns" appears to have been introduced.

A new circular dependency in the form of "evo/mnhftx -> validation -> evo/specialtxman -> evo/mnhftx" appears to have been introduced.

A new circular dependency in the form of "evo/specialtxman -> llmq/blockprocessor -> validation -> evo/specialtxman" appears to have been introduced.

Good job! The circular dependency "evo/specialtxman -> validation -> evo/specialtxman" is no longer present.
Please remove it from EXPECTED_CIRCULAR_DEPENDENCIES in ./test/lint/lint-circular-dependencies.sh
to make sure this circular dependency is not accidentally reintroduced.

Good job! The circular dependency "evo/chainhelper -> evo/specialtxman -> validation -> evo/chainhelper" is no longer present.
Please remove it from EXPECTED_CIRCULAR_DEPENDENCIES in ./test/lint/lint-circular-dependencies.sh
to make sure this circular dependency is not accidentally reintroduced.
```

Reply:
I think that's ok

Reply:
It is as expected. These circulate dependencies have a longer path and they just had been hidden by the shorter one.

The logic of script shows only the shortest one and by removing it you won't introduce any new circular dependencies but quite opposite reveal already existing. That's expected and fine to have. They can be addressed now one by one.

Reply:
The validation header was not added in this PR (see [`develop`](https://github.com/dashpay/dash/blob/c23514dc49c0a41fcafa8279a99d6d6d6c31a846/src/evo/specialtxman.cpp#L20)) so I'll remove the header and update circular dependencies in a separate PR, see https://github.com/dashpay/dash/pull/5941